src/Core/Migration/V6_7/Migration1739355802FixVatHandling.php

```diff
@@ -0,0 +1,72 @@
+<?php declare(strict_types=1);
+
+namespace Shopware\Core\Migration\V6_7;
+
+use Doctrine\DBAL\Connection;
+use Shopware\Core\Framework\Log\Package;
+use Shopware\Core\Framework\Migration\MigrationStep;
+
+/**
+ * @internal
+ *
+ * @codeCoverageIgnore
+ */
+#[Package('framework')]
+class Migration1739355802FixVatHandling extends MigrationStep
+{
+    /**
+     * @var array<string, string>
+     */
+    private array $countryIds;
+
+    public function getCreationTimestamp(): int
+    {
+        return 1739355802;
+    }
+
+    public function update(Connection $connection): void
+    {
+        $this->addCountryVatPattern($connection);
+    }
+
+    private function addCountryVatPattern(Connection $connection): void
+    {
+        $this->fetchCountryIds($connection);
+
+        foreach ($this->getCountryVatPattern() as $isoCode => $countryVatPattern) {
+            if (!\array_key_exists($isoCode, $this->countryIds)) {
+                // country was deleted by shop owner
+                continue;
+            }
+
+            $connection->update('country', ['vat_id_pattern' => $countryVatPattern], ['id' => $this->countryIds[$isoCode]]);
+        }
+    }
+
+    private function fetchCountryIds(Connection $connection): void
+    {
+        /** @var list<array{id: string, iso: string}> $countries */
+        $countries = $connection->executeQuery('SELECT `id`, `iso` FROM `country`')->fetchAllAssociative();
```

Comment:
Since you know all the iso's you want to adjust, doesn't it make sense to prefilter instead of fetching them all? If so, you can also iterate over the result instead and spare the `array_key_exists' as deleted ones will not be in the result